### DISCUSSION TITLE
Is Keycloak affected by CVE-2022-45787

### DISCUSSION THREAD

**Discussion by magixyu** [keycloak/keycloak#16472] [Q&A] [answered]
CVE-2022-45787
Description
Unproper laxist permissions on the temporary files used by MIME4J TempFileStorageProvider may lead to information disclosure to other local users. This issue affects Apache James MIME4J version 0.8.8 and prior versions. We recommend users to upgrade to MIME4j version 0.8.9 or later.

References
https://nvd.nist.gov/vuln/detail/CVE-2022-45787
https://lists.apache.org/thread/26s8p9stl1z261c4qw15bsq03tt7t0rj

It looks that ckey doesnot use the vulnerable function. Is it correct?

**Comment by sschu** [comment]
Not sure. But maybe that dependency can even be removed, at least the compile works without it.

**Reply by magixyu** [reply]
Thank you for your reply. I searched in the code, the "TempFileStorageProvider" is not used in kc.

**Comment by abstractj** [comment, marked as answer]
@magixyu it was fixed by https://github.com/keycloak/keycloak/issues/17556. Our team recently updated most of its dependency and this should be reflected by scanning the nightly images, and they should be part of upcoming releases:
```
trivy image --ignore-unfixed --timeout 25m quay.io/keycloak/keycloak:nightly
```
